File: android_java_hummus/src/main/java/com/github/klee0kai/hummus/adapterdelegates/CompositeAdapter.java
package com.github.klee0kai.hummus.adapterdelegates;

import com.hannesdorfmann.adapterdelegates4.AdapterDelegate;
import com.hannesdorfmann.adapterdelegates4.AdapterDelegatesManager;
import com.hannesdorfmann.adapterdelegates4.ListDelegationAdapter;

import java.util.List;

public class CompositeAdapter extends ListDelegationAdapter<List<Object>> {

    public CompositeAdapter() {
    }

    public CompositeAdapter(AdapterDelegatesManager<List<Object>> delegatesManager) {
        super(delegatesManager);
    }

    public static CompositeAdapter create(AdapterDelegate<List<Object>>... adapterDelegates) {
        AdapterDelegatesManager<List<Object>> manager = new AdapterDelegatesManager<>();
        for (AdapterDelegate<List<Object>> adapterDelegate : adapterDelegates)
            manager.addDelegate(adapterDelegate);
        return new CompositeAdapter(manager);
    }

}

File: android_kotlin_hummus/src/main/kotlin/com/github/klee0kai/hummus/adapterdelegates/CompositeAdapter.kt
package com.github.klee0kai.hummus.adapterdelegates

import com.hannesdorfmann.adapterdelegates4.AdapterDelegate
import com.hannesdorfmann.adapterdelegates4.AdapterDelegatesManager
import com.hannesdorfmann.adapterdelegates4.ListDelegationAdapter

open class CompositeAdapter(
    delegatesManager: AdapterDelegatesManager<List<Any>>
) : ListDelegationAdapter<List<Any>>(delegatesManager) {

    companion object {

        fun create(vararg adapterDelegates: AdapterDelegate<List<Any>>): CompositeAdapter =
            CompositeAdapter(
                AdapterDelegatesManager<List<Any>>().apply {
                    adapterDelegates.onEach {
                        addDelegate(it)
                    }
                }
            )

    }
}